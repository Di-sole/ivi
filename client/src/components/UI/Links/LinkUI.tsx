import { ReactNode } from 'react';
import styles from './LinkUI.module.scss';

interface LinkUIProps {
    children: ReactNode;
    linkTo?: string;
    shape: 'rectangular' | 'square' | 'round';
}

const LinkUI = ({ children, linkTo, shape }: LinkUIProps) => {
    const linkFormClass =
        shape === 'round'
            ? styles.link__wrapper_round
            : shape === 'square'
            ? styles.link__wrapper_square
            : styles.link__wrapper_rectangular;

    return (
        <a
            href={linkTo}
            className={[styles.link__wrapper, linkFormClass].join(' ')}
            target="blank"
            role="link-to-media"
        >
            <div className={styles.link__content}>{children}</div>
        </a>
    );
};

export default LinkUI;
