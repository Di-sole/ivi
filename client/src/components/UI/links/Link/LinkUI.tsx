import { ReactNode } from 'react';
import Link from 'next/link';
import styles from './LinkUI.module.scss';

interface LinkUIProps {
    children: ReactNode;
    href: string;
    shape: 'rectangular' | 'square' | 'round';
    className?: string;
}

const LinkUI = ({ children, href, shape, className }: LinkUIProps) => {
    const linkFormClass =
        shape === 'round'
            ? styles.container__wrapper_round
            : shape === 'square'
            ? styles.container__wrapper_square
            : styles.container__wrapper_rectangular;

    return (
        <Link
            href={href}
            className={[styles.container__wrapper, linkFormClass, className].join(' ')}
            target="blank"
            role="link-to-media"
        >
            <div className={styles.container__content}>{children}</div>
        </Link>
    );
};

export default LinkUI;
