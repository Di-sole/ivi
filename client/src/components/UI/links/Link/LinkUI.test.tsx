import React from 'react';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import Image from 'next/image';
import LinkUI from './LinkUI';
import './LinkUI.module.scss';
import footerStyles from '../../../footer/Footer.module.scss';
import devicesIcon from '../../../../public/icons/link_icons/devices_all.svg';

describe('LinkUI SNAPSHOT TESTS', () => {
    test('Round link should not be changed', () => {
        render(
            <LinkUI href="https://t.me/official_iviru" shape="round">
                <Image
                    src="https://solea-parent.dfs.ivi.ru/picture/ffffff,ffffff/social_telegram.svg"
                    height={16}
                    width={16}
                    alt="Логотип Telegram"
                />
            </LinkUI>,
        );
        const link = screen.getByRole('link-to-media');

        expect(link).toMatchSnapshot();
    });

    test('Rectangular link should not be changed', () => {
        render(
            <LinkUI href="#!" shape="rectangular" className={footerStyles.iviFooter__storeLink}>
                <div className={footerStyles.iviFooter__storeLink_textSingle}>
                    <Image src={devicesIcon} height={20} width={20} alt="Иконка умных устройств" />
                    Все устройства
                </div>
            </LinkUI>,
        );
        const link = screen.getByRole('link-to-media');

        expect(link).toMatchSnapshot();
    });
});
