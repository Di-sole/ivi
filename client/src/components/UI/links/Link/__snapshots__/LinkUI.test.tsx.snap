// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkUI SNAPSHOT TESTS Rectangular link should not be changed 1`] = `
<a
  class="link__wrapper link__wrapper_rectangular iviFooter__storeLink"
  href="#!"
  role="link-to-media"
  target="blank"
>
  <div
    class="link__content"
  >
    <div
      class="iviFooter__storeLink_textSingle"
    >
      <img
        alt="Иконка умных устройств"
        data-nimg="1"
        decoding="async"
        height="20"
        loading="lazy"
        src="/_next/image?url=%2Fimg.jpg&w=48&q=75"
        srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=48&q=75 2x"
        style="color: transparent;"
        width="20"
      />
      Все устройства
    </div>
  </div>
</a>
`;

exports[`LinkUI SNAPSHOT TESTS Round link should not be changed 1`] = `
<a
  class="link__wrapper link__wrapper_round "
  href="https://t.me/official_iviru"
  role="link-to-media"
  target="blank"
>
  <div
    class="link__content"
  >
    <img
      alt="Логотип Telegram"
      data-nimg="1"
      decoding="async"
      height="16"
      loading="lazy"
      src="https://solea-parent.dfs.ivi.ru/picture/ffffff,ffffff/social_telegram.svg"
      style="color: transparent;"
      width="16"
    />
  </div>
</a>
`;
